src/Temporalio.Extensions.Hosting/TemporalHostingServiceCollectionExtensions.cs

```diff
@@ -62,23 +63,96 @@ public static ITemporalWorkerServiceOptionsBuilder AddHostedTemporalWorker(
         public static ITemporalWorkerServiceOptionsBuilder AddHostedTemporalWorker(
             this IServiceCollection services, string taskQueue, string? buildId = null)
         {
+            // TODO: Review - We probably need to deprecate buildId param & add deployment version?
             // We have to use AddSingleton instead of AddHostedService because the latter does
             // not allow us to register multiple of the same type, see
             // https://github.com/dotnet/runtime/issues/38751.
             services.AddSingleton<IHostedService>(provider =>
                 ActivatorUtilities.CreateInstance<TemporalWorkerService>(
-                    provider, (TaskQueue: taskQueue, BuildId: buildId)));
+                    provider, new TemporalWorkerServiceIdentifier(taskQueue, buildId, true)));
             return new TemporalWorkerServiceOptionsBuilder(taskQueue, buildId, services).ConfigureOptions(
                 options =>
                 {
                     options.TaskQueue = taskQueue;
+#pragma warning disable 0618
                     options.BuildId = buildId;
+#pragma warning restore 0618
```

Comment:
If this is deprecated, does that mean that there's a way to use `WorkerDeploymentOptions` to represent "just build ID"? Otherwise, how do I represent just build ID in a non-deprecated way if I don't care about deployment name and won't ever enable versioning? And if I can do that, why does this not just call that overload?

Reply:
Yes, it means that, just Build ID is deprecated. You use a fully-formed deployment name now

Reply:
I think this is a bit rough on users who shouldn't have to care about deployment name to just have their existing build ID functionality remain